test/controllers/visit.controller.test.js

```diff
@@ -0,0 +1,131 @@
+import mongoose from 'mongoose';
+import { expect, request, useDatabase } from '../config';
+import User from '../../server/models/user.model';
+import VisitFactory from '../factories/visit.factory';
+import PropertyFactory from '../factories/property.factory';
+import UserFactory from '../factories/user.factory';
+import { addUser } from '../../server/services/user.service';
+import { addProperty } from '../../server/services/property.service';
+
+useDatabase();
+
+let userToken;
+
+const userId = mongoose.Types.ObjectId();
+const systemUser = UserFactory.build({ _id: userId, role: 0, activated: true });
+
+beforeEach(async () => {
+  userToken = await addUser(systemUser);
+});
+
+describe('Schedule Visit Route', () => {
+  const propId = mongoose.Types.ObjectId();
+  const demoProperty = PropertyFactory.build({ _id: propId, addedBy: userId, updatedBy: userId });
+
+  beforeEach(async () => {
+    await addProperty(demoProperty);
+  });
+
+  context('with valid data', () => {
+    it('returns successful property', (done) => {
+      const booking = VisitFactory.build({ propertyId: propId });
+      request()
+        .post('/api/v1/visit/book')
+        .set('authorization', userToken)
+        .send(booking)
+        .end((err, res) => {
+          expect(res).to.have.status(201);
+          expect(res.body.success).to.be.eql(true);
+          expect(res.body.message).to.be.eql('Visit scheduled successfully');
+          expect(res.body).to.have.property('schedule');
+          done();
+        });
+    });
+  });
+
+  context('when user token is used', () => {
+    beforeEach(async () => {
+      await User.findByIdAndDelete(userId);
+    });
+    it('returns token error', (done) => {
+      const booking = VisitFactory.build({ propertyId: propId });
+      request()
+        .post('/api/v1/visit/book')
+        .set('authorization', userToken)
```

Comment:
This should be invalid Token. I believe the endpoint is for users, so a user token should be able to consume the endpoint with no error. So you need to create an invalidToken to test this.
```suggestion
        .set('authorization', invalidToken)
```

Reply:
no this endpoint is for when a used token (when the user is deleted or no longer exists). Would add the suggested one tho